include/operations/SymbolicOperation.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "Definitions.hpp"
+#include "Expression.hpp"
+#include "StandardOperation.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <optional>
+#include <unordered_set>
+#include <variant>
+
+namespace qc {
+    // Overload pattern for std::visit
+    template<typename... Ts>
+    struct Overload: Ts... {
+        using Ts::operator()...;
+    };
+    template<class... Ts>
+    Overload(Ts...) -> Overload<Ts...>;
+
+    class SymbolicOperation final: public StandardOperation {
+    protected:
+        std::array<std::optional<Symbolic>, MAX_PARAMETERS> symbolicParameter{{std::nullopt, std::nullopt, std::nullopt}};
+
+        static OpType parseU3(const Symbolic& lambda, dd::fp& phi, dd::fp& theta);
+        static OpType parseU3(dd::fp& lambda, const Symbolic& phi, dd::fp& theta);
+        static OpType parseU3(dd::fp& lambda, dd::fp& phi, const Symbolic& theta);
+        static OpType parseU3(const Symbolic& lambda, const Symbolic& phi, dd::fp& theta);
+        static OpType parseU3(const Symbolic& lambda, dd::fp& phi, const Symbolic& theta);
+        static OpType parseU3(dd::fp& lambda, const Symbolic& phi, const Symbolic& theta);
+
+        static OpType parseU2(const Symbolic& lambda, const Symbolic& phi);
+        static OpType parseU2(const Symbolic& lambda, dd::fp& phi);
+        static OpType parseU2(dd::fp& lambda, const Symbolic& phi);
+
+        static OpType parseU1(const Symbolic& lambda);
+
+        void checkUgate();
+
+        void storeSymbolOrNumber(const SymbolOrNumber& param, std::size_t i);
+
+        [[nodiscard]] bool isSymbolicParameter(std::size_t i) const {
```

Comment:
```suggestion
        [[nodiscard]] bool isSymbolicParameter(const std::size_t i) const {
```

Reply:
Haven't had a const in a while. Good refresher.